public_html/wp-content/plugins/pattern-directory/includes/pattern-validation.php

```diff
@@ -200,5 +204,158 @@ function validate_status( $prepared_post, $request ) {
 		);
 	}
 
+	// Do not allow for non-privledged users to move a pending post to another status.
+	if (
+		'pending' === $current_status &&
+		'pending' !== $target_status &&
+		! current_user_can( $post_type->cap->edit_others_patterns )
+	) {
+		return new \WP_Error(
+			'rest_pattern_invalid_status',
+			sprintf(
+				__( 'Invalid post status. Status must be %s.', 'wporg-patterns' ),
+				'pending'
+			),
+			array( 'status' => 400 )
+		);
+	}
+
+	return $prepared_post;
+}
+
+/**
+ * Validate the pattern doesn't appear to be spam.
+ */
+function validate_against_spam( $prepared_post, $request ) {
+	if ( is_wp_error( $prepared_post ) ) {
+		return $prepared_post;
+	}
+
+	$target_status = isset( $request['status'] ) ? $request['status'] : '';
+
+	// Only run spam checks at publish time.
+	if ( 'publish' !== $target_status ) {
+		return $prepared_post;
+	}
+
+	// Extract strings and URLs, run against spam checks.
+	$post = get_post( $prepared_post->ID );
+
+	$title       = $prepared_post->post_title ?? $post->post_title;
+	$content     = $prepared_post->post_content ?? $post->post_content;
+	$description = $request['meta']['wpop_description'] ?? ( $post->wpop_description ?: '' );
+	$keywords    = $request['meta']['wpop_keywords'] ?? ( $post->wpop_keywords ?: '' );
+
+	// Extract URLs.
+	$links = array();
+	if ( preg_match_all( '![\b\W"\'](?P<link>((http|https|ftp|mailto):)?//[/.\w]+)[\b\W"\']!', $content, $m ) ) {
+		$links = array_unique( $m['link'] );
+	}
+
+	// Stringify.
+	if ( ! class_exists( '\WordPressdotorg\Pattern_Translations\Pattern' ) ) {
+		// This is just a fall-back for local environments where the Translator isn't active.
+		// not designed to be used in production.
+		$strings = array(
+			$title,
+			$description,
+			wp_strip_all_tags( $content ),
+			$keywords,
+		);
+	} else {
+		$pattern              = new Translations_Pattern();
+		$pattern->ID          = $post->ID;
+		$pattern->title       = $title;
+		$pattern->name        = $post->post_name;
+		$pattern->description = $description;
+		$pattern->keywords    = $keywords;
+		$pattern->html        = $content;
+		$pattern->locale      = get_locale();
+
+		$parser  = new Translations_PatternParser( $pattern );
+		$strings = $parser->to_strings();
+	}
+
+	// Combine strings for ease of use.
+	$combined_strings = implode( "\n", $strings );
+	$combined_links   = implode( "\n", $links );
+	$combined         = $combined_strings . "\n" . $combined_links;
```

Comment:
I don't see `$combined` used, should this be sent to akismet instead of just `$combined_strings`?

Reply:
Yeah, `$combined` isn't being used, nor is `$links` / `$combined_links`.

I had added those with the intention of using them, but never did.. 

I'll pull those out for now.